xla/service/while_loop_unroller.h

```diff
@@ -117,6 +126,12 @@ class WhileLoopUnroller : public HloModulePass {
                                      bool wrap_in_trivial_loop = false,
                                      bool force_unroll = false,
                                      bool prepare = true);
+  // Similar to above, but returns the unrolled flag and
+  // the new unrolled while instruction.
+  static absl::StatusOr<UnrollResult> UnrollAndReturnReplacement(
```

Comment:
I don't think it's worth adding a new method. The existing Unroll method is only used [here](https://github.com/openxla/xla/blob/0fbf2e1eef1c45ade8ebc8e070b734d81112a30c/xla/service/hlo_unstacker.cc#L1422). Instead just return `UnrollResult` from Unroll and change the single current usage.

Reply:
im hesitant to do that, this pass is not used anywhere in OSS but it's being actively maintained, so probably some other internal compilers are using this pass? this function is a public static function.